extensions/email/src/main/java/org/mule/extension/email/internal/commands/StoreCommand.java

```diff
@@ -0,0 +1,126 @@
+
+    //TODO: annotated the parameter directory with @Path when available
+
+    /**
+     * Stores the specified email of id {@code emailId} into the configured {@code directory}.
+     * <p>
+     * if no emailId is specified, the operation will try to find an email or {@link List} of emails
+     * in the incoming {@link MuleMessage}.
+     * <p>
+     * If no email(s) are found in the {@link MuleMessage} and no {@code emailId} is specified.
+     * the operation will fail.
+     * <p>
+     * The emails are stored as mime message in a ".txt" format.
+     * <p>
+     * The name of the email file is composed by the subject and
+     * the received date of the email.
+     *
+     * @param connection  the associated {@link RetrieverConnection}.
+     * @param muleMessage the incoming {@link MuleMessage}.
+     * @param folderName  the name of the folder where the email(s) is going to be fetched.
+     * @param directory   the directory where the emails are going to be stored.
+     * @param emailId     the optional number of the email to be marked. for default the email is taken from the incoming {@link MuleMessage}.
+     */
+    public void store(RetrieverConnection connection, MuleMessage muleMessage, String folderName, String directory, Integer emailId)
+    {
+        try
+        {
+            Folder folder = getOpenFolder(folderName, READ_ONLY, connection.getStore());
+            if (emailId == null)
+            {
+                Object payload = muleMessage.getPayload();
+                if (payload instanceof List)
+                {
+                    for (Object o : (List) payload)
+                    {
+                        if (o instanceof MuleMessage)
+                        {
+                            emailId = getIdOrFail(((MuleMessage) o));
+                            storeMessage(emailId, folder, directory);
+                        }
+                        else
+                        {
+                            throw new EmailException("Cannot perform operation for the incoming payload");
+                        }
+                    }
+                    return;
+                }
+                emailId = getIdOrFail(muleMessage);
+            }
+            storeMessage(emailId, folder, directory);
+            closeFolder(folder, false);
+        }
+        catch (MessagingException | IOException me)
+        {
+            throw new EmailRetrieverException(me);
+        }
+    }
+
+    /**
+     * Stores a single {@link Message} into the specified {@code directory}.
+     *
+     * @param emailId   the emailId corresponding to the message that is going to be stored
+     * @param folder    the folder to look for the email
+     * @param directory the directory where the message is going to be stored
+     */
+    private void storeMessage(int emailId, Folder folder, String directory) throws IOException, MessagingException
```

Comment:
that's not a good experience. add an `overwrite:boolean` parameter which defaults to false. If the file exists and the flag is false then fail. Don't overwrite files without the user's concent